core/src/main/java/com/sap/ai/sdk/core/AiCoreDeployment.java

```diff
@@ -0,0 +1,224 @@
+  }
+
+  /**
+   * Create a new instance of the AI Core service with a deployment.
+   *
+   * @param service The AI Core service.
+   * @param scenarioId The scenario id.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public static AiCoreDeployment forScenarioId(
+      @Nonnull final AiCoreService service, @Nonnull final String scenarioId) {
+    final Predicate<AiDeployment> p = deployment -> scenarioId.equals(deployment.getScenarioId());
+    return new AiCoreDeployment(service, obj -> obj.getDeploymentId(service.client(), p));
+  }
+
+  /**
+   * Create a new instance of the AI Core service with a deployment.
+   *
+   * @param service The AI Core service.
+   * @param deploymentId The deployment id.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public static AiCoreDeployment forDeploymentId(
+      @Nonnull final AiCoreService service, @Nonnull final String deploymentId) {
+    return new AiCoreDeployment(service, obj -> deploymentId);
+  }
+
+  @Nonnull
+  @Override
+  public Destination destination() {
+    final var dest = service.baseDestinationHandler.apply(service);
+    final var builder = service.builderHandler.apply(service, dest);
+    destinationSetUrl(builder, dest);
+    destinationSetHeaders(builder, dest);
+    return builder.build();
+  }
+
+  @Nonnull
+  @Override
+  public ApiClient client() {
+    final var destination = destination();
+    return service.clientHandler.apply(service, destination);
+  }
+
+  /**
+   * Update and set the URL for the destination.
+   *
+   * @param builder The destination builder.
+   * @param dest The original destination reference.
+   */
+  protected void destinationSetUrl(
+      @Nonnull final DefaultHttpDestination.Builder builder, @Nonnull final Destination dest) {
+    String uri = dest.get(DestinationProperty.URI).get();
+    if (!uri.endsWith("/")) {
+      uri = uri + "/";
+    }
+    builder.uri(uri + "v2/inference/deployments/%s/".formatted(getDeploymentId()));
+  }
+
+  /**
+   * Update and set the default request headers for the destination.
+   *
+   * @param builder The destination builder.
+   * @param dest The original destination reference.
+   */
+  protected void destinationSetHeaders(
+      @Nonnull final DefaultHttpDestination.Builder builder, @Nonnull final Destination dest) {
+    builder.property(AI_RESOURCE_GROUP, getResourceGroup());
+  }
+
+  /**
+   * Set the resource group.
+   *
+   * @param resourceGroup The resource group.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public AiCoreDeployment withResourceGroup(@Nonnull final String resourceGroup) {
+    return new AiCoreDeployment(service, deploymentId, resourceGroup);
+  }
+
+  /**
+   * Set the destination.
+   *
+   * @param destination The destination.
+   * @return A new instance of the AI Core service.
+   */
+  @Nonnull
+  public AiCoreDeployment withDestination(@Nonnull final Destination destination) {
```

Comment:
Do we need this? If yes we should test that this works like expected.
Also I can't understand from the Javadoc, please update .

Reply:
This method could be useful when retroactively changing the base destination.
```java
new AiCoreService()
  .forDeploymentByScenario("orchestration")
  .withDestination( mockedDestination )
  .client();
```

But I understand this is not a realistic scenario currently, when the following remains possible:
```java
new AiCoreService()
  .withDestination( mockedDestination )
  .forDeploymentByScenario("orchestration")
  .client();
```